controllers/agents/svc/serviceclass_controller.go

```diff
@@ -130,15 +145,20 @@ func (r *ServiceClassReconciler) Reconcile(ctx context.Context, req ctrl.Request
 			// fallthrough: we still want to write the service class status field
 		}
 	} else if controllerutil.ContainsFinalizer(&serviceClass, finalizer) {
-		handler := func(remote_client client.Client, rs v1alpha1.RegisteredService) error {
+		handler := func(remote_client client.Client, rs v1alpha1.RegisteredService, secret *v1.Secret) error {
 			if err := remote_client.Delete(ctx, &rs); err != nil {
 				if apierrors.IsNotFound(err) {
 					// we tried to delete an object that doesn't exist, so
 					return nil
 				}
 				reconcileLog.Error(err, "Failed to delete registered service", "namespace", rs.Namespace)
+				return err
 			}
-			return err
+
+			if secret != nil {
```

Comment:
We are setting the Registered Service as owner of the secret in L130, so can deletion of secret be avoided here?

Reply:
Yep, good catch.  Removed.